PSReadLine/ConsoleKeyChordConverter.cs

```diff
@@ -243,7 +246,20 @@ static bool MapKeyChar(string input, ref ConsoleModifiers mods, out ConsoleKey k
                 else
                 {
                     key = 0;
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Is this a Windows only problem? Not using `VkKeyScan` at initialization time (e.g. registering a binding) was important because it doesn't handle keyboard layout changes at all.

Reply:
Hmm, it's a windows only problem for this particular binding: <kbd>Ctrl+[</kbd> --> `CommandMode in VI`, but <kbd>Ctrl+[</kbd> cannot be bound in general for other actions.

This is because on Linux, <kbd>Ctrl+[</kbd> produces `CosnoleKeyInfo  { KeyChar = \u0016, Key = Escape, Modifiers = 0 }`, which maps to `PSKeyInfo { Escape }`. `PSKeyInfo { Escape }` is already bound with `CommandMode` in VI. However, when running `Set-PSReadLineKeyHandler -Chord 'Ctrl+[' -Function ...`, the string `'Ctrl+['` results in `CosnoleKeyInfo { KeyChar = \u0016, Key = 0, Modifiers = Control }`, which maps to `PSKeyInfo { Control+Escape }`, so it will never be hit when pressing <kbd>Ctrl+[</kbd>.

Similarly on Windows, the chord string `'Ctrl+['` generates `PSKeyInfo { Control+Escape }`, but <kbd>Ctrl+[</kbd> produces `ConsoleKeyInfo { KeyChar = \u0016, Key = oem4, Modifiers = Control }`, which results in `PSKeyInfo { Control+[ }`, so the binding will never be hit.

The current change in this PR is not right. This needs to be fixed on all platforms.

Reply:
> Not using VkKeyScan at initialization time (e.g. registering a binding) was important because it doesn't handle keyboard layout changes at all.

Does it really matter in that scenario? If the user choose to register a binding in one keyboard layout and then switch to another keyboard layout, then I won't expect the registered key binding to work, because the same key chord is likely to be mapped to a different `PSKeyInfo` anyways ...

Reply:
If I bind <kbd>q</kbd> in `en-US`, should that binding not work when changing to `fr-FR`? The same applies to any key like <kbd>Ctrl+[</kbd> - we shouldn't be using the scan code at binding time.

Now the fact that `VkKeyScan` doesn't work on layout changes means that it doesn't really matter, but that's a bug that could get fixed someday.

Reply:
I will play with the "registering binding in one layout and use in another layout" a bit more to better understand what might be the right thing to do.
But for <kbd>Ctrl+[</kbd> no being able to bind to custom actions in general on Linux, what's your thought about it? Shall we keep the current behavior (<kbd>Ctrl+[</kbd> basically equivalent to <kbd>Esc</kbd>)?

Reply:
My general thoughts in this area are:
* Don't try to emulate things that don't make sense - that code doesn't need to be written.
* If something can't be supported easily, *consider* adding a warning/error.
* It's OK to silently do something a little weird for corner cases.